onset_maproom/maproom.py

```diff
@@ -422,5 +431,127 @@ def cess_plots(
         )
         return cessDate_graph, probExceed_cess, tab_style, None
 
+
+# TODO can we make this any more redundant?
+@APP.callback(
+    Output("onset_layer", "url"),
+    Input("search_start_day", "value"),
+    Input("search_start_month", "value"),
+    Input("searchDays", "value"),
+    Input("wetThreshold", "value"),
+    Input("runningDays", "value"),
+    Input("runningTotal", "value"),
+    Input("minRainyDays", "value"),
+    Input("dryDays", "value"),
+    Input("drySpell", "value"),
+)
+def onset_tile_url(
+        search_start_day,
+        search_start_month,
+        search_days,
+        wet_thresh,
+        wet_spell_length,
+        wet_spell_thresh,
+        min_wet_days,
+        dry_spell_length,
+        dry_spell_search,
+):
+    qstr = urllib.parse.urlencode({
+        "search_start_day": search_start_day,
+        "search_start_month": search_start_month,
+        "search_days": search_days,
+        "wet_thresh": wet_thresh,
+        "wet_spell_length": wet_spell_length,
+        "wet_spell_thresh": wet_spell_thresh,
+        "min_wet_days": min_wet_days,
+        "dry_spell_length": dry_spell_length,
+        "dry_spell_search": dry_spell_search,
+    })
+    return f"{TILE_PFX}/mean/{{z}}/{{x}}/{{y}}?{qstr}"
+
+
+@SERVER.route(f"{TILE_PFX}/mean/<int:tz>/<int:tx>/<int:ty>")
+def onset_tile(tz, tx, ty):
+    parse_arg = pingrid.parse_arg
+    search_start_day = parse_arg("search_start_day", int)
+    search_start_month1 = parse_arg("search_start_month", calc.strftimeb2int)
+    search_days = parse_arg("search_days", int)
+    wet_thresh = parse_arg("wet_thresh", float)
+    wet_spell_length = parse_arg("wet_spell_length", int)
+    wet_spell_thresh = parse_arg("wet_spell_thresh", float)
+    min_wet_days = parse_arg("min_wet_days", int)
+    dry_spell_length = parse_arg("dry_spell_length", int)
+    dry_spell_search = parse_arg("dry_spell_search", int)
```

Comment:
@remicousin I wanted to point out that I'm using a different argument processing paradigm here than what we did for other tile URLs. The calculation parameters go in the query part of the URL, e.g.
```
http://foo.bar/baz?search_start_day=1&search_start_month=Jan
```
instead of in the path part of the URL, and we use `pingrid.parse_arg` instead of flask's built in argument processing to parse them and do type conversion (e.g. string to int). This makes the URLs more readable and easier to test, and frees us from flask's sometimes strange rules (e.g. the problem you had where "90" wasn't recognized as a float).